pkg/build/builder/common.go

```diff
@@ -252,6 +252,12 @@ func HandleBuildStatusUpdate(build *buildapiv1.Build, client buildclientv1.Build
 		if latestBuild == nil {
 			latestBuild, err = client.Get(build.Name, metav1.GetOptions{})
 			if err != nil {
+				latestBuild = nil
+				return false, nil
+			}
+			if latestBuild.Name == "" {
```

Comment:
If for whatever reason we don't get a Name, I think it's safe to assume the rest of the object is empty or invalid. In either case we should re-fetch the build object.